### ISSUE TITLE
[Bug][fhe-ai] train_encrypted is simulated - loss hardcoded to 0.5 and weights never updated

### ISSUE THREAD

**Issue by ionfwsrijan** [KanishJebaMathewM/Truxify#10807]
## Summary

`FHETrainer.train_encrypted` is a simulated loop: the reported loss is hardcoded to `0.5` every epoch, no gradients are computed, and no weights are ever updated. Training via `POST /fhe-ai/model/train` performs zero learning.

## Root cause

`backend/fhe-ai/fhe_model.py:183-207`:

```python
# Simple training (simulated)
losses = []
for epoch in range(epochs):
    # Forward pass (encrypted)
    y_pred = self.model.forward(X_enc)

    # Loss (simulated)
    loss = 0.5
    losses.append(loss)
```

The route `POST /fhe-ai/model/train` (routes.py:41-56) returns `{'success': True, 'data': {'losses': [0.5, 0.5, ...], 'final_loss': 0.5}}`, giving callers a training curve that never changes regardless of data, epochs, or model architecture.

## Impact

- Any workflow that uses this endpoint to train an encrypted model gets a constant `0.5` loss and unchanged weights — the returned model has the same random initialization it had before "training".
- Combined with `_encrypted_linear` ignoring weights (see related), prediction quality cannot improve through this API.

## Suggested fix

Implement real (even if simplified) encrypted gradient descent, or remove/disable the endpoint and fail closed rather than returning a fabricated training report.
